### PR TITLE
LWDEV-10293 Remove duplication of code in RedisPersistenceManager and RedisWithoutOrdersPersistenceManager

### PR DESCRIPTION
**Pull request by rocketmannot**

### CODE REVIEW

File: src/main/kotlin/com/lykke/matching/engine/database/redis/RedisPersistenceManager.kt [hunk 1]
```diff
@@ -20,21 +19,20 @@ import com.lykke.utils.logging.MetricsLogger
 import org.apache.log4j.Logger
 import org.springframework.util.CollectionUtils
 import redis.clients.jedis.Transaction
+import redis.clients.jedis.exceptions.JedisException
 
 class RedisPersistenceManager(
```
[the remark this answers is not in the document]
Comment (barsukovskii) [reply src/main/kotlin/com/lykke/matching/engine/database/redis/RedisPersistenceManager.kt]: Yes, currently it is so. But we consider a task to switch off trusted clients events - in this case there will be a lot of these cases (persistence without sequence numbers) and we will have to change back. Well, let's leave it as is currently, but please don't change any logic in the frame of refactoring task next time

File: src/main/kotlin/com/lykke/matching/engine/database/common/strategy/AzurePersistOrdersStrategy.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package com.lykke.matching.engine.database.common.strategy
+
+import com.lykke.matching.engine.database.common.entity.OrderBooksPersistenceData
+import com.lykke.matching.engine.holders.OrdersDatabaseAccessorsHolder
+import com.lykke.matching.engine.holders.StopOrdersDatabaseAccessorsHolder
+import redis.clients.jedis.Transaction
+
+class AzurePersistOrdersStrategy(private val ordersDatabaseAccessorsHolder: OrdersDatabaseAccessorsHolder,
```
Comment (barsukovskii) [inline src/main/kotlin/com/lykke/matching/engine/database/common/strategy/AzurePersistOrdersStrategy.kt]: We don't persist orders to azure - only files or redis

Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/database/common/strategy/AzurePersistOrdersStrategy.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/database/common/strategy/OrdersPersistInSecondaryDbStrategyFactory.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Autowired
+    private lateinit var persistedStopApplicationEventPublisher: SimpleApplicationEventPublisher<StopOrderBookPersistEvent>
+
+    @Autowired
+    private lateinit var config: Config
+
+    override fun getObject(): OrdersPersistInSecondaryDbStrategy? {
+        if (config.me.storage == Storage.Redis || config.me.storage == Storage.RedisWithoutOrders) {
+            return AzureSecondaryDbOrderPersistStrategy(persistedOrdersApplicationEventPublisher, persistedStopApplicationEventPublisher)
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/database/common/strategy/OrdersPersistInSecondaryDbStrategyFactory.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/database/common/strategy/OrdersPersistInSecondaryDbStrategyFactory.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    private lateinit var persistedOrdersApplicationEventPublisher: SimpleApplicationEventPublisher<OrderBookPersistEvent>
+
+    @Autowired
+    private lateinit var persistedStopApplicationEventPublisher: SimpleApplicationEventPublisher<StopOrderBookPersistEvent>
+
+    @Autowired
+    private lateinit var config: Config
+
+    override fun getObject(): OrdersPersistInSecondaryDbStrategy? {
+        if (config.me.storage == Storage.Redis || config.me.storage == Storage.RedisWithoutOrders) {
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/database/common/strategy/OrdersPersistInSecondaryDbStrategyFactory.kt]: fixed

**Pull request merged**